### ISSUE TITLE
helpers.CCSubcribe$4.onGroupActionMessageReceived

### ISSUE THREAD

**Issue by joseBarreto** [CometChat/xamarin-forms-chat-sdk-demo#21]
Hello,
We're having problems grouping functionality for Android.

Whenever a user is added to a group, the application closes with these errors:

java.lang.NoSuchMethodError: No virtual method subscribeToTopic(Ljava/lang/String;)Lcom/google/android/gms/tasks/Task; in class Lcom/google/firebase/messaging/FirebaseMessaging; or its super classes (declaration of 'com.google.firebase.messaging.FirebaseMessaging' appears in /data/app/com.fieldcorp.Lagoinha-wW4SHhq5LT0LUGn6vwEjSA==/base.apk:classes4.dex)

helpers.CCSubcribe$4.onGroupActionMessageReceived CCSubcribe.java:558
com.inscripts.helpers.MessageHelper.handleOneOnOneMessage MessageHelper.java:504
com.inscripts.heartbeats.CCHeartbeat.processReceiveData CCHeartbeat.java:762
com.inscripts.heartbeats.CCHeartbeat.access$100 CCHeartbeat.java:47
com.inscripts.heartbeats.CCHeartbeat$1$1.doInBackground CCHeartbeat.java:150
com.inscripts.heartbeats.CCHeartbeat$1$1.doInBackground CCHeartbeat.java:147
android.os.AsyncTask$2.call AsyncTask.java:333
java.util.concurrent.FutureTask.run FutureTask.java:266
android.os.AsyncTask$SerialExecutor$1.run AsyncTask.java:245
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1162
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:636
java.lang.Thread.run Thread.java:764

**Comment by joseBarreto** [comment]
Java.Lang.NoSuchMethodError: No virtual method subscribeToTopic(Ljava/lang/String;)Lcom/google/android/gms/tasks/Task; in class Lcom/google/firebase/messaging/FirebaseMessaging; or its super classes (declaration of 'com.google.firebase.messaging.FirebaseMessaging' appears in /data/app/com.fieldcorp.Lagoinha-gkxLj25MOEnGdEvL-2LQCg==/base.apk:classes4.dex)

**Comment by mehtanishant** [comment]
Hello,

We are trying to replicate the mentioned issue at our end in our sample app. Kindly let us know if you are using any other Firebase dependencies in your project.

Thank You.

**Comment by joseBarreto** [comment]
> Hello,
> 
> We are trying to replicate the mentioned issue at our end in our sample app. Kindly let us know if you are using any other Firebase dependencies in your project.
> 
> Thank You.

No, since the CometChat package installs the Firebase dependencies on 60+ versions, I removed mine from my project.
The only 'code' I use for the Firebase is or token.


This guy.
 [Service(Permission = "android.permission.BIND_JOB_SERVICE", Exported = true)]
    [IntentFilter(new[] { "com.google.firebase.INSTANCE_ID_EVENT" })]
    public class ConnectFirebase : FirebaseInstanceIdService
    {
        private const string TAG = "MyFirebaseIIDService";
        public override void OnTokenRefresh()
        {
            var refreshedToken = FirebaseInstanceId.Instance.Token;
            Log.Debug(TAG, "Refreshed token: " + refreshedToken);
        }

        private void SendRegistrationToServer(string token)
        {
            // Add custom implementation, as needed.
        }
    }

**Comment by mehtanishant** [comment]
Hello,

Thank you for providing us more information. We are working on this at our end. We will keep you posted about the progress.
